app/services/booking_responder.rb

```diff
@@ -84,4 +86,17 @@ def mark_banned_visitors
       visitor.update! banned: true
     end
   end
+
+  def create_message
+    return nil if booking_response.message_body.blank?
+
+    Message.create!(body: booking_response.message_body,
+                    user: booking_response.user,
+                    visit: visit,
+                    visit_state_change: visit.last_visit_state)
```

Comment:
This is a bit nasty. Although I can see that currently you're calling this method inside a transaction which also changes the visit state, this won't always be the case when creating messages, and seems rather non-obvious. Can it be more explicit?

I still think it would be cleaner if acceptance or rejections were first class objects which may have associated messages, and associated state changes – rather than tying the message to the state change directly.